jans-cedarling/cedarling/src/jwt/decoding_strategy/key_service/error.rs

```diff
@@ -0,0 +1,36 @@
+/*
+ * This software is available under the Apache-2.0 license.
+ * See https://www.apache.org/licenses/LICENSE-2.0.txt for full text.
+ *
+ * Copyright (c) 2024, Gluu, Inc.
+ */
+
+/// Error type for the Key Service
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    /// Indicates that a key with the specified `kid` was not found in the JWKS.
+    #[error("No key with `kid`=\"{0}\" found in the JWKS.")]
```

Comment:
What is the external name for a key id?

Reply:
im not sure what you mean by external name but JSON Web Keys could optionally have a `kid` field which refers to the *key id* of the key.